README.md

```diff
@@ -322,6 +322,101 @@ export default {
+into the engine's namespace instead of the host application.
+
+Beyond that it adds in a configuration module for the engine, and nothing else. 
+It is a remarkably straightforward process.
+
+### Lazy Engines
+
+Lazy engines are built in the same way as eager engines, but their assets are
+not combined back into the host application's `vendor.js` file. This means that
+they are run through a separate and unique build process from what a default
+addon will go through, though it reaches out to the upstream implementation in
+Ember CLI where possible.
+
+A lazy engine's output (`lazy-engine`) looks like this:
+```
+dist
+├── assets
+│   ├── host-application.css
+│   ├── host-application.js
+│   ├── vendor.css
+│   └── vendor.js
+├── crossdomain.xml
+├── engines-dist
+│   └── lazy-engine
+│       ├── engine.js
+│       ├── engine.css
+│       ├── engine-vendor.css
+│       ├── engine-vendor.js
+│       └── public-asset.jpg
+├── index.html
+└── robots.txt
+```
+
+#### /routes.js
+
+The routes.js and anything it imports must be present at boot time of the host
+application. It will be bundled into the host application's vendor.js file. This
+location should be considered `undefined` behavior and should not be relied upon
+as it may change in the future.
+
+Its module name inside of the host application will be `lazy-engine/routes`. Any
+imports will also be in the `lazy-engine` module path.
+
+#### /app
+
+Assets in this folder don't make sense and will be ignored as they break the
+isolation guarantees of engines.
+
+#### /addon
+
+JavaScript assets in this folder will be processed as per normal addon behavior
+except that they will end up inside of the `engine.js` file. Their module
+definition will be rooted to the engine name.
+
+For example, `/addon/routes/application.js` will result in a JavaScript module
+named `lazy-engine/routes/application` inside of the
+`/dist/engines-dist/lazy-engine/engine.js` file.
+
+#### /addon/templates
+
+Templates will be compiled by your engine but they must include
+`ember-cli-htmlbars` inside of `dependencies` in the engine's `package.json`. 
+
+As an example, `/addon/templates/application.hbs` will result in a JavaScript
+module named `lazy-engine/templates/application` inside of the
+`/dist/engines-dist/lazy-engine/engine.js` file.
+
+#### /addon/styles/addon.css
+
+FIXME!
+
+#### /addon/styles/(other).css
+
+FIXME!
+
+#### /public
+
+Assets appearing in the public folder will appear at the root of the engine
+output with no transformation. For example `/public/public-asset.jpg` appears at
+the root level of the `/engines-dist/lazy-engine/` output folder. Assets in this
+folder have no default behavior and you are responsible for handling their use.
+
+#### Asset Manifest
+
+Further, the engine must enumerate its primary assets (JS and CSS) in order to
+be loaded by the asset loading service. That will be generated at
+`/dist/lazy-engine/asset-manifest.json` at build time. It will also by default
```

Comment:
Yes, since hooks don't run in nested addons, we'll rely on the manifest being generated from the `ember-engines` addon installed in the host application.